dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -383,7 +386,23 @@ private LambdaEvent createBasicEvent(String repository, String gitReference, Str
             final SourceFile dockstoreYml) {
         try {
             List<Workflow> updatedWorkflows = new ArrayList<>();
+            final Path gitRefPath = Path.of(gitReference);
             for (YamlWorkflow wf : yamlWorkflows) {
+                final List<String> filters = wf.getFilter();
+                // Ignore filters if none specified
+                Boolean isFiltered = !filters.isEmpty();
+                for (String filter : filters) {
+                    final PathMatcher pathMatcher = FileSystems.getDefault().getPathMatcher("glob:**/" + filter);
```

Comment:
Have not worked with this library yet (looks like a good choice though)
Does  this `**/` mean that if someone specifies `develop` as a filter will it will match a branch called `feature/develop`?
Alternatively, will it be hubflow compatible? 
For example if someone specifies `feature/foo` as a filter, will this work or would it only match `potato/feature/foo`?

Reply:
If someone specified `develop`, then any reference path ending in `develop` would match.
If you wanted to only match the exact develop branch, then to be safe using `heads/develop` or even better the full `refs/heads/develop` path will work.
The only caveat is if for whatever reason you have branchs called something like `feature/refs/heads/develop`.

Reply:
Ok, open to discussion with others on #dockstore. 
What I would expect is that `develop` would match only `develop` and that one would need to use `*develop` or `foo/develop` to match `foo/develop`

Similarly, I would expect `feature/foo` to match `feature/foo` but not `potato/feature/foo`

I guess the simpler way to explain would be that I would expect to have to specify a `*` to use globbing in the filter and that otherwise it would be an exact match. In fact, looking around I think the convention is to use slashes to surround regular expressions otherwise exact match. Sorry for not catching that in the original ticket. 



Possible documentation examples
https://www.appveyor.com/docs/branches/
```
Regular expressions should be surrounded by /, otherwise Appveyor will do simple case insensitive string comparison.
```
https://circleci.com/docs/2.0/configuration-reference/#filters-1
```
Branches can have the keys only and ignore which either map to a single string naming a branch. You may also use regular expressions to match against branches by enclosing them with slashes, or map to a list of such strings. Regular expressions must match the entire string.
```

https://docs.travis-ci.com/user/customizing-the-build#safelisting-or-blocklisting-branches
```
Any name surrounded with / in the list of branches is treated as a regular expression and can contain any quantifiers, anchors or character classes supported by Ruby regular expressions.
```

Reply:
Hm, this made me realize that `**/` actually should be `**` to support the complete `refs/heads/develop` since there's no leading slash... time for a 1-character commit

Reply:
Perhaps a simple solution to this implementation-wise could be trying 2 separate matchers where instead of `**/` leading the glob, we have:
- `refs/heads/`
- `refs/tags/`
This way it's enforced that the branch/tag is always an exact match to some tag or branch.

Optionally there can also be `refs/` to allow for specifying a tag or a branch.

Reply:
It is worth a try. 
Looking through these pages, it seems some require the user to specify whether the filter is for a tag or for a branch. Some seem to assume that names of tags cannot overlap with branches with is probably true in 99% of cases

Reply:
Added this in latest commit